S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java

```diff
@@ -357,12 +366,52 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         }
     }
 
+
+    private void  requestPermission()
+    {
+        //ask for the permission in android M
+        int permission = ContextCompat.checkSelfPermission(this,
+                Manifest.permission.WRITE_EXTERNAL_STORAGE);
+
+        if (permission != PackageManager.PERMISSION_GRANTED) {
+            Log.i(TAG, "Permission to record denied");
+
+            if (ActivityCompat.shouldShowRequestPermissionRationale(this,
+                    Manifest.permission.WRITE_EXTERNAL_STORAGE)) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(this);
+                builder.setMessage("Permission to access the SD-CARD is required for this app to Download PDF.")
```

Comment:
change message to: "Permission to access the External Storage is required for this application to store the downloaded data from Amazon S3"

Reply:
done